doc/can-route.md

```diff
@@ -173,82 +189,119 @@ route.data.update( { page: "tasks", id: 5 } );
 You will see this result in the URL and `location.hash`.
 
-```js
-route.data.type = "image/bar";
-// OR
-route.attr( "type", "image/bar" );
-```
-
-The URL will look like this:
-
-    https://example.com/#!type=image%2Fbar
-
-The location hash will look like this:
+```html
+<mock-url></mock-url>
+<script type="module">
+import "//unpkg.com/mock-url@^5.0.0";
+import {DefineMap, route} from "can";
 
-    #!type=image%2Fbar
+route.data = new DefineMap( {type: "image/bar"} ); // location.hash -> #!&type=image%2Fbar
+route.start();
+</script>
+```
+@codepen
 
 ## Creating a route
 
-Use `route.register(url, defaults)` to create a
+Use [`route.register(url, defaults)`](can-route.register) to create a
 routing rule. A rule is a mapping from a url to
 an object (that is the route’s data).
 In order to map to specific properties in the url,
 prepend a colon to the name of the property like:
 
-```js
-route.register( "#!content/{type}" );
+```html
+<mock-url></mock-url>
+<script type="module">
+import "//unpkg.com/mock-url@^5.0.0";
+import {route} from "can";
+
+route.register( "content/{type}" );
+route.data.type = "example"; // location.hash -> #!content/example
+route.start();
+</script>
 ```
+@codepen
 
 If no routes are added, or no route is matched,
-can-route’s data is updated with the [can-route.deparam deparamed]
+can-route’s data is updated with the [can-route.deparam deparam]ed
 hash.
 
-```js
-location.hash = "#!type=videos";
+```html
+<mock-url></mock-url>
+<script type="module">
+import "//unpkg.com/mock-url@^5.0.0";
+import {route} from "can";
 
-// route -> {type : "videos"}
+location.hash = "#!&type=videos";
+route.start();
+
+console.log(route.data); //-> {type : "videos"}
+</script>
 ```
+@codepen
 
 Once routes are added and the hash changes,
 can-route looks for matching routes and uses them
 to update can-route’s data.
 
-```js
-route.register( "#!content/{type}" );
-location.hash = "#!content/images";
+```html
+<mock-url></mock-url>
+<script type="module">
+import "//unpkg.com/mock-url@^5.0.0";
+import {route} from "can";
 
-// route -> {type : "images"}
-route.data.type = "songs";
+route.register( "content/{type}" );
+location.hash = "#!content/images";
+route.start();
 
-// location.hash -> "#!content/songs"
+console.log( route.data ) //-> {type : "images"}
+route.data.type = "songs"; // location.hash -> "#!content/songs"
+</script>
 ```
+@codepen
 
 Default values can be added to a route:
 
-```js
-route.register( "content/{type}", { type: "videos" } );
+```html
+<mock-url></mock-url>
+<script type="module">
+import "//unpkg.com/mock-url@^5.0.0";
+import {route} from "can";
+
+route.register( "content/{type}", {type: "videos"} );
 location.hash = "#!content/";
 
-// route -> {type : "videos"}
+route.start();
+
+console.log( route.data ); //-> {type: "videos"}
 // location.hash -> "#!content/"
+</script>
 ```
+@codepen
 
 Defaults can also be set on the root page of your app:
```

Comment:
mention the empty string (`""`) is treated as the  "root" page of your app.  If there is no hash (or if using pushstate, someone is at `/`.